api/src/main/java/com/epam/pipeline/security/saml/SAMLUserDetailsServiceImpl.java

```diff
@@ -93,16 +95,32 @@ public UserContext loadUserBySAML(SAMLCredential credential) {
             return userContext;
         } else {
             LOGGER.debug("Found user by name {}", userName);
+            if (loadedUser.isBlocked()) {
+                LOGGER.debug("User {} is blocked!", userName);
+                throw new AuthenticationException("User is blocked!");
+            }
             loadedUser.setUserName(userName);
             List<Long> roles = loadedUser.getRoles().stream().map(Role::getId).collect(Collectors.toList());
             if (userManager.needToUpdateUser(groups, attributes, loadedUser)) {
                 loadedUser = userManager.updateUserSAMLInfo(loadedUser.getId(), userName, roles, groups, attributes);
                 LOGGER.debug("Updated user groups {} ", groups);
             }
+
+            if (anyGroupIsBlocked(groups)) {
+                LOGGER.debug("User {} is blocked due to one of his groups is blocked!", userName);
+                throw new AuthenticationException("User is blocked!");
+            }
             return new UserContext(loadedUser);
         }
     }
 
+    private boolean anyGroupIsBlocked(final List<String> groups) {
+        return groups.stream().anyMatch(s -> {
+            final GroupStatus status = userManager.loadNullableGroupBlockingStatus(s);
```

Comment:
To reduce a number of requests to DB, let's introduce a method to load status by `List<String>`